src/backend/fs.rs

```diff
@@ -138,15 +143,16 @@ pub async fn write_catalog(
 
     let write_pk_str = write_pk_str.to_owned();
     let file_hash = file_hash.to_string();
+    let name = name.unwrap_or(file_hash);
```

Comment:
so unwrap_or, will use file_hash if name is_none()
That is what I think here. 
This is what you discussed giving the user the option to name a file.

Reply:
Correct.